packages/commons/src/fileManager.ts

```diff
@@ -1,110 +1,115 @@
 import {
   CopyObjectCommand,
   DeleteObjectCommand,
+  ListObjectsCommand,
+  PutObjectCommand,
   S3Client,
+  S3ClientConfig,
 } from "@aws-sdk/client-s3";
 import { FileManagerConfig } from "./config/fileManagerConfig.js";
 import { logger } from "./index.js";
 
 export type FileManager = {
-  deleteFile: (container: string, path: string) => Promise<void>;
+  delete: (bucket: string, path: string) => Promise<void>;
   copy: (
-    container: string,
-    path: string,
+    bucket: string,
     filePathToCopy: string,
-    documentId: string,
+    path: string,
+    resourceId: string,
     fileName: string
   ) => Promise<string>;
   storeBytes: (
-    documentId: string,
-    documentName: string,
-    byteArray: Uint8Array
+    bucket: string,
+    path: string,
+    resourceId: string,
+    fileName: string,
+    fileContent: Buffer
   ) => Promise<string>;
+  listFiles: (bucket: string) => Promise<string[]>;
 };
 
-const mockFileManager: FileManager = {
-  deleteFile: async (container: string, path: string): Promise<void> => {
-    logger.info(`Deleting file ${path} from container ${container}`);
-
-    return Promise.resolve();
-  },
-
-  copy: async (
-    container: string,
-    _path: string,
-    filePathToCopy: string,
-    _documentId: string,
-    _fileName: string
-  ): Promise<string> => {
-    logger.info(
-      `Mock Copying file ${filePathToCopy} from container ${container}`
-    );
-    return Promise.resolve("");
-  },
-  storeBytes: async (
-    _documentId: string,
-    _documentName: string,
-    _byteArray: Uint8Array
-  ): Promise<string> => {
-    logger.info(`Mock Storing bytes`);
-    return Promise.resolve("mock/path/to/file");
-  },
-};
-
-export function initFileManager(config: FileManagerConfig): FileManager {
-  if (config.mockFileManager) {
-    return mockFileManager;
-  } else {
-    const { s3AccessKeyId, s3SecretAccessKey, s3Region } = config;
+export function initFileManager(
+  config: FileManagerConfig,
+  debugLogs?: boolean
+): FileManager {
+  const s3ClientConfig: S3ClientConfig = {
+    credentials: {
+      accessKeyId: config.s3AccessKeyId,
+      secretAccessKey: config.s3SecretAccessKey,
+    },
+    region: config.s3Region,
+    endpoint: config.s3LocalMinio
+      ? `${config.s3LocalMinioHost}:${config.s3LocalMinioPort}`
+      : undefined,
+    forcePathStyle: config.s3LocalMinio ? true : false,
+    logger: debugLogs ? console : undefined,
```

Comment:
possible future improvement: we could use winston `logger.debug` here and pilot log level directly from global configs

Reply:
I tried to use `logger.debug` or `logger.info` but it does not print anything for some reason 🥲 Since it is a debug option, I did not investigate it too much and went with console